src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -80,10 +96,80 @@ public record VectorSearchConfig(
                 implementation = String.class)
             @JsonProperty("metric")
             @JsonAlias("function") // old name
-            String metric) {
-      public VectorSearchConfig(Integer dimension, String metric) {
+            String metric,
+        @Valid
+            @Nullable
+            @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "Optional vectorize configuration to provide embedding service",
+                type = SchemaType.OBJECT,
+                implementation = VectorizeConfig.class)
+            @JsonProperty("service")
+            VectorizeConfig vectorizeConfig) {
+
+      public VectorSearchConfig(Integer dimension, String metric, VectorizeConfig vectorizeConfig) {
         this.dimension = dimension;
         this.metric = metric == null ? "cosine" : metric;
+        this.vectorizeConfig = vectorizeConfig;
+      }
+
+      public record VectorizeConfig(
```

Comment:
Is this supposed to be included?

Reply:
Just to align with the ddl configuration in objectId spec.
Shouldn't be an issue, since we have ready expose vectorize in prod.